crates/ark/src/modules/positron/graphics.R

```diff
@@ -40,59 +40,35 @@ render_directory <- function() {
     directory
 }
 
-render_path <- function(id) {
+render_path <- function(id, format) {
     directory <- render_directory()
-    file <- paste0("render-", id, ".png")
+    file <- paste0("render-", id, ".", format)
     file.path(directory, file)
 }
 
 #' @export
 .ps.graphics.create_device <- function() {
     name <- "Ark Graphics Device"
 
-    # TODO: Remove the "shadow" device in favor of implementing our own
-    # minimal graphics device like {devoid}. That would allow us to remove
-    # all of the awkwardness here around:
-    # - A `filename` that we never look at
-    # - A `res` that isn't scaled by `pixel_ratio`
-    # - The fact that the `png` device is forcing double the work to happen,
-    #   as it is drawing graphics that we never look at.
-    # - The fact that `locator()` doesn't work b/c `png` doesn't support it.
-    directory <- render_directory()
-    filename <- file.path(directory, "current-plot.png")
-    type <- default_device_type()
-    res <- default_resolution_in_pixels_per_inch()
-
-    # Create the graphics device that we are going to shadow
-    withCallingHandlers(
-        grDevices::png(
-            filename = filename,
-            type = type,
-            res = res
-        ),
-        warning = function(w) {
-            stop("Error creating graphics device: ", conditionMessage(w))
-        }
-    )
+    # Create the graphics device that we are going to shadow.
+    # Creating a graphics device mutates global state, we don't need to capture
+    # the return value.
+    device_shadow()
 
     # Update the device name + description in the base environment.
     index <- grDevices::dev.cur()
-    oldDevice <- .Devices[[index]]
-    newDevice <- name
+    old_device <- .Devices[[index]]
+    new_device <- name
 
     # Copy device attributes. Usually, this is just the file path.
-    attributes(newDevice) <- attributes(oldDevice)
-
-    # Set other device properties.
-    attr(newDevice, "type") <- type
-    attr(newDevice, "res") <- res
```

Comment:
Reasonably certain these "other device properties" are not useful

Reply:
Agreed - they are not part of the spec and only set on some devices